19-pied-raven/technical_specification.md

```diff
@@ -0,0 +1,64 @@
+# File Encryption & Decryption Service Integration (Pied Crow)
+**Epic Type:** Implementation Epic
+
+
+## Scope:
+A scope definition can be found here: https://wiki.verbis.dkfz.de/x/4AC2D
+## User Journeys
+
+Building on the [*18 - Hooded Crow*](../18-hooded-crow/technical_specification.md) epic, this epic aims to finish the work started on the encryption & decryption services, integrating them into the wider GHGA microservice landscape and resolving remaining questions and issues.
+
+### 1. Enable Encryption Key Store service to use HashiCorp Vault for secure storage and retrieval of file secrets
+File secrets are currently stored in a MongoDB attached to the Encryption Key Store.
+For safe storage concerns this should be replaced with HashiCorp Vault instead.
+The API should allow to store a secret and return a newly generated ID by which this secret can be retrieved.
+
+Implementing this change includes multiple steps:
+
+1. Eplore how to set up HashiCorp Vault for our use case
+2. Explore how to connect, store and retrieve secrets
+3. Provide a synchronous API for communication with the Vault
+
+### 2. Integrate with and adapt adjacent services
+Both the Encryption Key Store and the Interrogation Room depend on (synchronous and asynchronous) communication with other services.
+Interaction with the respective service on side of the caller/receiver is not implemented yet.
+Check and adjust all of the following services:
+
+1. Upload Controller - outbound event signaling file upload completed to Interrogation Room
+2. Upload Controller - handling of failure event from Interrogation Room
+3. Internal File Registry - handling of success event from Interrogation Room
+4. Download Controller - request and receive envelope, attach envelope to outgoing file cotent
```

Comment:
Maybe, we can define this as optional. First integrating and testing IRS and EKSS would be sufficient.

Reply:
Originally, I would have spoken against that, but with the refactoring of the other services currently going on, we should wait for them to be finished, before adding functionality.